### PR TITLE
fix(iam_policy_no_administrative_privileges): check attached policies and AWS-Managed

### LINKED ISSUE #1917: Make iam_policy_no_administrative_privileges check configurable to scan AWS managed policies
**Is your feature request related to a problem? Please describe.**
Due to the way `iam_service.py` retrieves policies using the `Local` scope (https://github.com/prowler-cloud/prowler/blob/7008d2a953f94dcca21432fa89a77522967dfc98/prowler/providers/aws/services/iam/iam_service.py#L344), the `iam_policy_no_administrative_privileges` check only raises findings if a customer managed policy with "*:*" permissions is attached to an IAM principal, as opposed to an AWS managed policy such as `arn:aws:iam::aws:policy/AdministratorAccess`.

While this behavior can be useful in many scenarios, the CIS 1.5 benchmark specifies that no "*:*" policies should be attached to any IAM principals, regardless of whether they are AWS managed. 

**Describe the solution you'd like**
Make the iam_policy_no_administrative_privileges check configurable so that a user can specify that AWS managed policies should be scanned by this check as well.

**Describe alternatives you've considered**

**Additional context**
While scanning AWS managed policies is useful for compliance enforcement, it is not necessarily desirable default behavior as it can result in a large number of inactionable findings.

### PR DESCRIPTION
**Pull request by MrCloudSec**
### Description

- Create IAM Policy BaseModel class.
- Add the following checks:
1. `iam_customer_unattached_policy_no_administrative_privileges`: renamed current check `iam_policy_no_administrative_privileges` to change its behavior that looks for not attached ones in the description and title. LOW, to fix it is deleting it or change it. Add into the description that may be eventual consistent if an ephemeral resource is using it.
2. `iam_customer_attached_policy_no_administrative_privileges`: that checks only customer managed policies attached. HIGH, to fix it is changing it.
3. `iam_aws_attached_policy_no_administrative_privileges`: that checks only AWS managed IAM policies attached. HIGH, to fix it is detaching it.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by toniblyx** [comment]
Currently we have `iam_policy_no_administrative_privileges` that checks customer managed policies attached and not attached. This one doesn't belong to CIS 1.4 or 1.5 for AWS. So we can leave it as it is.
Now, we need two more new checks to fit the CIS requirements:
- `iam_customer_managed_attached_policy_no_administrative_privileges`: that checks only customer managed policies attached.
- `iam_aws_managed_policy_attached_no_administrative_privileges`:  that checks only aws managed policies attached.

**Comment by MrCloudSec** [comment]
> Currently we have `iam_policy_no_administrative_privileges` that checks customer managed policies attached and not attached. This one doesn't belong to CIS 1.4 or 1.5 for AWS. So we can leave it as it is. Now, we need to more new checks to fit the CIS requirements:
> 
> * `iam_customer_managed_attached_policy_no_administrative_privileges`: that checks only customer managed policies attached.
> * `iam_aws_managed_policy_attached_no_administrative_privileges`:  that checks only aws managed policies attached.

I would change the check name `iam_policy_no_administrative_privileges` to `iam_customer_policy_no_administrative_privileges` and create a new check called `iam_policy_attached_no_administrative_privileges` which will be the one mapping CIS.

**Comment by toniblyx** [comment]
- `iam_policy_no_administrative_privileges`: rename this one to  `iam_customer_unattached_policy_no_administrative_privileges` change its behavior that looks for not attached ones in the description and title. LOW, to fix it is deleting it or change it. Add into the description that may be eventual consistent if an ephemeral resource is using it.
- `iam_customer_attached_policy_no_administrative_privileges`: that checks only customer managed policies attached. HIGH, to fix it is changing it.
- `iam_aws_attached_policy_no_administrative_privileges`: that checks only AWS managed IAM policies attached. HIGH, to fix it is detaching it.

File: prowler/compliance/aws/aws_foundational_security_best_practices_aws.json [hunk 1]
```diff
@@ -362,14 +362,16 @@
         }
       ],
       "Checks": [
-        "iam_policy_no_administrative_privileges",
+        "iam_aws_attached_policy_no_administrative_privileges",
+        "iam_customer_attached_policy_no_administrative_privileges",
```
Comment (MrCloudSec) [inline prowler/compliance/aws/aws_foundational_security_best_practices_aws.json]:
```suggestion
```

File: prowler/compliance/aws/fedramp_low_revision_4_aws.json [hunk 2]
```diff
@@ -26,9 +26,11 @@
         "opensearch_service_domains_cloudwatch_logging_enabled",
         "guardduty_is_enabled",
         "iam_password_policy_minimum_length_14",
-        "iam_policy_no_administrative_privileges",
+        "iam_aws_attached_policy_no_administrative_privileges",
+        "iam_customer_attached_policy_no_administrative_privileges",
```
Comment (MrCloudSec) [inline prowler/compliance/aws/fedramp_low_revision_4_aws.json]:
```suggestion
```

File: prowler/compliance/aws/fedramp_low_revision_4_aws.json [hunk 3]
```diff
@@ -58,9 +60,11 @@
         "ec2_instance_public_ip",
         "ec2_instance_imdsv2_enabled",
         "emr_cluster_master_nodes_no_public_ip",
-        "iam_policy_no_administrative_privileges",
+        "iam_aws_attached_policy_no_administrative_privileges",
+        "iam_customer_attached_policy_no_administrative_privileges",
```
Comment (MrCloudSec) [inline prowler/compliance/aws/fedramp_low_revision_4_aws.json]:
```suggestion
```

File: prowler/compliance/aws/fedramp_moderate_revision_4_aws.json [hunk 4]
```diff
@@ -91,9 +92,11 @@
       ],
       "Checks": [
         "iam_password_policy_minimum_length_14",
-        "iam_policy_no_administrative_privileges",
+        "iam_aws_attached_policy_no_administrative_privileges",
+        "iam_customer_attached_policy_no_administrative_privileges",
```
Comment (MrCloudSec) [inline prowler/compliance/aws/fedramp_moderate_revision_4_aws.json]:
```suggestion
```

File: prowler/compliance/aws/fedramp_moderate_revision_4_aws.json [hunk 5]
```diff
@@ -144,9 +147,11 @@
...
```
Comment (MrCloudSec) [inline prowler/compliance/aws/fedramp_moderate_revision_4_aws.json]:
```suggestion
```

File: prowler/compliance/aws/fedramp_moderate_revision_4_aws.json [hunk 6]
```diff
@@ -188,9 +193,11 @@
         "ec2_instance_public_ip",
         "ec2_instance_imdsv2_enabled",
         "emr_cluster_master_nodes_no_public_ip",
-        "iam_policy_no_administrative_privileges",
+        "iam_aws_attached_policy_no_administrative_privileges",
+        "iam_customer_attached_policy_no_administrative_privileges",
```
Comment (MrCloudSec) [inline prowler/compliance/aws/fedramp_moderate_revision_4_aws.json]:
```suggestion
```

File: prowler/compliance/aws/fedramp_moderate_revision_4_aws.json [hunk 7]
```diff
@@ -249,9 +256,11 @@
       ],
       "Checks": [
         "iam_password_policy_minimum_length_14",
-        "iam_policy_no_administrative_privileges",
+        "iam_aws_attached_policy_no_administrative_privileges",
+        "iam_customer_attached_policy_no_administrative_privileges",
```
Comment (MrCloudSec) [inline prowler/compliance/aws/fedramp_moderate_revision_4_aws.json]:
```suggestion
```

File: prowler/compliance/aws/fedramp_moderate_revision_4_aws.json [hunk 8]
```diff
@@ -269,8 +278,10 @@
         }
       ],
       "Checks": [
-        "iam_policy_no_administrative_privileges",
-        "iam_policy_no_administrative_privileges",
+        "iam_aws_attached_policy_no_administrative_privileges",
+        "iam_customer_attached_policy_no_administrative_privileges",
```
Comment (MrCloudSec) [inline prowler/compliance/aws/fedramp_moderate_revision_4_aws.json:274]:
```suggestion
```

File: prowler/compliance/aws/ffiec_aws.json [hunk 9]
```diff
@@ -418,9 +418,11 @@
       ],
       "Checks": [
         "ec2_instance_profile_attached",
-        "iam_policy_no_administrative_privileges",
+        "iam_aws_attached_policy_no_administrative_privileges",
+        "iam_customer_attached_policy_no_administrative_privileges",
```
Comment (MrCloudSec) [inline prowler/compliance/aws/ffiec_aws.json]:
```suggestion
```

File: prowler/compliance/aws/ffiec_aws.json [hunk 10]
```diff
@@ -519,8 +521,10 @@
         }
       ],
       "Checks": [
-        "iam_policy_no_administrative_privileges",
-        "iam_policy_no_administrative_privileges"
+        "iam_aws_attached_policy_no_administrative_privileges",
+        "iam_customer_attached_policy_no_administrative_privileges",
```
Comment (MrCloudSec) [inline prowler/compliance/aws/ffiec_aws.json]:
```suggestion
```

File: prowler/compliance/aws/ffiec_aws.json [hunk 11]
```diff
@@ -536,8 +540,10 @@
...
```
Comment (MrCloudSec) [inline prowler/compliance/aws/ffiec_aws.json]:
```suggestion
```

File: prowler/compliance/aws/ffiec_aws.json [hunk 12]
```diff
@@ -553,8 +559,10 @@
         }
       ],
       "Checks": [
-        "iam_policy_no_administrative_privileges",
-        "iam_policy_no_administrative_privileges",
+        "iam_aws_attached_policy_no_administrative_privileges",
+        "iam_customer_attached_policy_no_administrative_privileges",
```
Comment (MrCloudSec) [inline prowler/compliance/aws/ffiec_aws.json:557]:
```suggestion
```

File: prowler/compliance/aws/ffiec_aws.json [hunk 13]
```diff
@@ -579,8 +587,10 @@
         "iam_password_policy_number",
         "iam_password_policy_symbol",
         "iam_password_policy_uppercase",
-        "iam_policy_no_administrative_privileges",
-        "iam_policy_no_administrative_privileges",
+        "iam_aws_attached_policy_no_administrative_privileges",
+        "iam_customer_attached_policy_no_administrative_privileges",
```
Comment (MrCloudSec) [inline prowler/compliance/aws/ffiec_aws.json:583]:
```suggestion
```

File: prowler/compliance/aws/ffiec_aws.json [hunk 14]
```diff
@@ -755,9 +765,11 @@
       "Checks": [
         "cloudtrail_multi_region_enabled",
         "cloudtrail_cloudwatch_logging_enabled",
-        "iam_policy_no_administrative_privileges",
+        "iam_aws_attached_policy_no_administrative_privileges",
+        "iam_customer_attached_policy_no_administrative_privileges",
```
Comment (MrCloudSec) [inline prowler/compliance/aws/ffiec_aws.json]:
```suggestion
```

File: prowler/compliance/aws/gxp_21_cfr_part_11_aws.json [hunk 15]
```diff
@@ -82,9 +82,11 @@
         "iam_password_policy_number",
         "iam_password_policy_symbol",
         "iam_password_policy_uppercase",
-        "iam_policy_no_administrative_privileges",
+        "iam_aws_attached_policy_no_administrative_privileges",
+        "iam_customer_attached_policy_no_administrative_privileges",
```
Comment (MrCloudSec) [inline prowler/compliance/aws/gxp_21_cfr_part_11_aws.json]:
```suggestion
```

File: prowler/compliance/aws/gxp_21_cfr_part_11_aws.json [hunk 16]
```diff
@@ -166,9 +168,11 @@
...
```
Comment (MrCloudSec) [inline prowler/compliance/aws/gxp_21_cfr_part_11_aws.json]:
```suggestion
```

File: prowler/providers/aws/services/iam/iam_no_custom_policy_permissive_role_assumption/iam_no_custom_policy_permissive_role_assumption.py
Comment (jfagoagas) [inline prowler/providers/aws/services/iam/iam_no_custom_policy_permissive_role_assumption/iam_no_custom_policy_permissive_role_assumption.py:1]: What are the changes in this check? Is difficult to follow with the diff.

File: prowler/providers/aws/services/iam/iam_policy_allows_privilege_escalation/iam_policy_allows_privilege_escalation.py [hunk 17]
```diff
@@ -61,63 +61,65 @@ def execute(self) -> Check_Report_AWS:
         }
         findings = []
         for policy in iam_client.policies:
-            report = Check_Report_AWS(self.metadata())
-            report.resource_id = policy["PolicyName"]
-            report.resource_arn = policy["Arn"]
-            report.region = iam_client.region
+            # Check only custom policies
+            if policy.type == "Custom":
```
Comment (jfagoagas) [inline prowler/providers/aws/services/iam/iam_policy_allows_privilege_escalation/iam_policy_allows_privilege_escalation.py:65]: Does not make sense to check all policies here right?

File: prowler/providers/aws/services/iam/iam_service.py [hunk 18]
```diff
@@ -366,7 +366,34 @@ def __list_policies__(self):
...
+                            Policy(
+                                name=policy["PolicyName"],
+                                arn=policy["Arn"],
+                                version_id=policy["DefaultVersionId"],
+                                type="AWS",
+                                attached=True
+                                if policy["AttachmentCount"] > 0
+                                else False,
+                            )
+                        )
```
Comment (jfagoagas) [inline prowler/providers/aws/services/iam/iam_service.py]: Cloud you please move this into a separate function? The only aspect that differs is the `scope` and you can pass it as argument and then use it in the AWS API call and under `policy.type`. Thanks!

File: tests/providers/aws/services/iam/iam_aws_attached_policy_no_administrative_privileges/iam_aws_attached_policy_no_administrative_privileges_test.py [hunk 19]
```diff
@@ -0,0 +1,165 @@
...
+            assumed_role_info=None,
+            audited_regions=["us-east-1", "eu-west-1"],
+            organizations_metadata=None,
+            audit_resources=None,
+        )
+
+        return audit_info
+
+    @mock_iam
+    def test_policy_administrative(self):
```
Comment (jfagoagas) [inline tests/providers/aws/services/iam/iam_aws_attached_policy_no_administrative_privileges/iam_aws_attached_policy_no_administrative_privileges_test.py]:
```suggestion
    def test_policy_with_administrative_privileges(self):
```

File: tests/providers/aws/services/iam/iam_aws_attached_policy_no_administrative_privileges/iam_aws_attached_policy_no_administrative_privileges_test.py
Comment (jfagoagas) [inline tests/providers/aws/services/iam/iam_aws_attached_policy_no_administrative_privileges/iam_aws_attached_policy_no_administrative_privileges_test.py:1]: Could you create the test case for 0 policies and findings?

Also assert the region and the tags in all the tests please.

File: tests/providers/aws/services/iam/iam_customer_attached_policy_no_administrative_privileges/iam_customer_attached_policy_no_administrative_privileges_test.py
Comment (jfagoagas) [inline tests/providers/aws/services/iam/iam_customer_attached_policy_no_administrative_privileges/iam_customer_attached_policy_no_administrative_privileges_test.py:1]: Could you create the test case for 0 policies and findings?

Also assert the region and the tags in all the tests please.

File: tests/providers/aws/services/iam/iam_customer_unattached_policy_no_administrative_privileges/iam_customer_unattached_policy_no_administrative_privileges_test.py
Comment (jfagoagas) [inline tests/providers/aws/services/iam/iam_customer_unattached_policy_no_administrative_privileges/iam_customer_unattached_policy_no_administrative_privileges_test.py:1]: Could you create the test case for 0 policies and findings?

Also assert the region and the tags in all the tests please.

File: prowler/providers/aws/services/iam/iam_service.py [hunk 20]
```diff
@@ -489,3 +516,12 @@ class Certificate(BaseModel):
     id: str
     arn: str
     expiration: datetime
+
+
+class Policy(BaseModel):
```
Comment (jfagoagas) [inline prowler/providers/aws/services/iam/iam_service.py:534]: We need to include the tags for the policies right?

**Review by jfagoagas — CHANGES_REQUESTED** [review]

File: prowler/providers/aws/services/iam/iam_service.py (hunk 20 above)
Comment (MrCloudSec) [reply prowler/providers/aws/services/iam/iam_service.py:534]: Done

File: prowler/providers/aws/services/iam/iam_service.py (hunk 18 above)
Comment (MrCloudSec) [reply prowler/providers/aws/services/iam/iam_service.py]: Done

**Review by jfagoagas — APPROVED** [review]

**Comment by jfagoagas** [comment]
Not related with the above PR, it was a typo.

**Pull request merged**